src/main/java/com/direwolf20/buildinggadgets/common/tools/ToolRenders.java

```diff
@@ -64,304 +72,175 @@ public static void updateInventoryCache() {
-        if (heldItem.isEmpty()) return;
+        // Calculate the players current position, which is needed later
+        Vec3d playerPos = ToolRenders.Utils.getPlayerTranslate(player, evt.getPartialTicks());
 
-        Minecraft mc = Minecraft.getMinecraft();
-        mc.renderEngine.bindTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
+        // Render if we have a remote inventory selected
+        renderLinkedInventoryOutline(heldItem, playerPos);
 
-        //Calculate the players current position, which is needed later
-        double doubleX = player.lastTickPosX + (player.posX - player.lastTickPosX) * evt.getPartialTicks();
-        double doubleY = player.lastTickPosY + (player.posY - player.lastTickPosY) * evt.getPartialTicks();
-        double doubleZ = player.lastTickPosZ + (player.posZ - player.lastTickPosZ) * evt.getPartialTicks();
+        RayTraceResult lookingAt = VectorTools.getLookingAt(player, heldItem);
+        IBlockState state = Blocks.AIR.getDefaultState();
+        List<BlockPos> coordinates = getAnchor(heldItem);
 
-        //Prepare the block rendering
-        BlockRendererDispatcher dispatcher = Minecraft.getMinecraft().getBlockRendererDispatcher();
-        BlockRenderLayer origLayer = MinecraftForgeClient.getRenderLayer();
+        if (lookingAt == null && coordinates.size() == 0)
+            return;
 
-        Integer dim = GadgetUtils.getDIMFromNBT(heldItem, "boundTE");
-        BlockPos pos = GadgetUtils.getPOSFromNBT(heldItem, "boundTE");
+        IBlockState startBlock = ToolRenders.Utils.getStartBlock(lookingAt, player);
+        if (startBlock == ModBlocks.effectBlock.getDefaultState())
+            return;
 
-        if (dim != null && pos != null) {
-            GlStateManager.pushMatrix();//Push matrix again just because
-            GlStateManager.enableBlend();
-            //This blend function allows you to use a constant alpha, which is defined later
-            GlStateManager.blendFunc(GL11.GL_CONSTANT_ALPHA, GL11.GL_ONE_MINUS_CONSTANT_ALPHA);
-            GlStateManager.translate(-doubleX, -doubleY, -doubleZ);//The render starts at the player, so we subtract the player coords and move the render to 0,0,0
-            GlStateManager.translate(pos.getX(), pos.getY(), pos.getZ());//Now move the render position to the coordinates we want to render at
-            GlStateManager.rotate(-90.0F, 0.0F, 1.0F, 0.0F); //Rotate it because i'm not sure why but we need to
-            GlStateManager.translate(-0.005f, -0.005f, 0.005f);
-            GlStateManager.scale(1.01f, 1.01f, 1.01f);
-            GL14.glBlendColor(1F, 1F, 1F, 0.35f); //Set the alpha of the blocks we are rendering
-            dispatcher.renderBlockBrightness(Blocks.STAINED_GLASS.getDefaultState().withProperty(COLOR, EnumDyeColor.YELLOW), 1f);
+        IBlockState renderBlockState = getToolBlock(heldItem);
+
+        //Don't render anything if there is no block selected (Air)
+        if (renderBlockState == Blocks.AIR.getDefaultState())
```

Comment:
Isn't there some isAir Method? (I'm sure that it's a thing in 1.13 because of the 3 air types - Does it exist in 1.12?)

Reply:
No :P